pubsub/test/iam.test.js

```diff
@@ -14,159 +14,278 @@
+        assert.equal(policy, undefined);
+      });
+    });
+    it('should handle error', function () {
+      var sample = getSample();
+      var error = 'error';
+      sample.mocks.topic.iam.testPermissions.callsArgWith(1, new Error(error));
+      sample.program.testTopicPermissions(topicName, function (err, permissions) {
+        assert(err);
+        assert(err.message === 'error');
+        assert.equal(permissions, undefined);
       });
     });
   });
 
-  describe('testSubscriptionPermissionsExample', function () {
-    it('handles error', function () {
-      var subscription = {
-        iam: {
-          testPermissions: sinon.stub().callsArgWith(1, 'error')
-        }
-      };
-      proxyquire('../iam', {
-        './subscription': {
-          pubsub: {
-            subscription: sinon.stub().returns(subscription)
-          }
-        }
-      }).testSubscriptionPermissionsExample('test-subscription', function (err) {
-        assert(err === 'error');
+  describe('testSubscriptionPermissions', function () {
+    it('should tests a subscription\'s permissions', function () {
+      var sample = getSample();
+      sample.program.testSubscriptionPermissions(subscriptionName, function (err, permissions) {
+        assert.ifError(err);
+        assert.equal(permissions, permissionsMock);
+        assert(console.log.calledWith('Tested permissions for: %s', subscriptionName));
       });
     });
+    it('should require subscriptionName', function () {
+      var sample = getSample();
+      sample.program.testSubscriptionPermissions(undefined, function (err, policy) {
+        assert(err);
+        assert(err.message = '"subscriptionName" is required!');
+        assert.equal(policy, undefined);
+      });
+    });
+    it('should handle error', function () {
+      var sample = getSample();
+      var error = 'error';
+      sample.mocks.subscription.iam.testPermissions.callsArgWith(1, new Error(error));
+      sample.program.testSubscriptionPermissions(subscriptionName, function (err, permissions) {
+        assert(err);
+        assert(err.message === 'error');
+        assert.equal(permissions, undefined);
+      });
+    });
+  });
+
+  describe('printUsage', function () {
+    it('should print usage', function () {
+      var program = getSample().program;
+
+      program.printUsage();
+
+      assert(console.log.calledWith('Usage: node iam RESOURCE COMMAND [ARGS...]'));
+      assert(console.log.calledWith('\nResources:\n'));
+      assert(console.log.calledWith('\ttopics'));
+      assert(console.log.calledWith('\tsubscriptions'));
+      assert(console.log.calledWith('\nCommands:\n'));
+      assert(console.log.calledWith('\tget NAME'));
+      assert(console.log.calledWith('\tset NAME'));
+      assert(console.log.calledWith('\ttest NAME'));
+      assert(console.log.calledWith('\nExamples:\n'));
+      assert(console.log.calledWith('\tnode iam topics get my-topic'));
```

Comment:
I don't suppose it's worth combining some of these lines? e.g. `node iam topics [get/set/test] my-topic`